rdr_service/dao/study_nph_dao.py

```diff
@@ -872,6 +872,18 @@ def get_event_by_source_name(self, source_name):
             )
             return records.first()
 
+    def get_event_by_name(self, name: str) -> List:
+        if not hasattr(self.model_type, 'name'):
+            return []
+
+        with self.session() as session:
+            records = session.query(
+                self.model_type
+            ).filter(
+                self.model_type.source_name == name
+            )
+            return records.first()
```

Comment:
The type hint for the method gives `List`, but wouldn't `records.first()` would return a model type object?